### PR TITLE
add distance in proto requests

### PR DESCRIPTION
**Pull request by danielnaumau**

### CODE REVIEW

File: src/main/java/com/traveltime/sdk/dto/requests/TimeFilterProtoDistanceRequest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        return TimeFilterFastRequestOuterClass.TimeFilterFastRequest
+                .newBuilder()
+                .setOneToManyRequest(oneToManyBuilder.build())
+                .build()
+                .toByteArray();
+    }
+
+    @Override
+    public Either<TravelTimeError, Request> createRequest(URI baseUri, TravelTimeCredentials credentials) {
+        String uri = "https://proto-with-distance.api.traveltimeapp.com/api/v2/uk/time-filter/fast/driving+ferry";
```
Comment (Donatas-L) [inline src/main/java/com/traveltime/sdk/dto/requests/TimeFilterProtoDistanceRequest.java:63]: We also support walking and more countries than UK, we also have IE

**Comment by Donatas-L** [comment]
You should also add support for multiple countries and multiple transportation modes for proto with distance.

**Comment by Donatas-L** [comment]
Right now we support UK, IE and driving, walking

**Review by mjanuszkiewicz-tt — APPROVED** [review]

**Review by Donatas-L — CHANGES_REQUESTED** [review]
Lacks support for different transportation modes and countries

File: src/main/java/com/traveltime/sdk/dto/requests/TimeFilterProtoDistanceRequest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import io.vavr.control.Either;
+import lombok.NonNull;
+import okhttp3.Request;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.RandomAccess;
+
+public class TimeFilterProtoDistanceRequest extends ProtoRequest {
```
Comment (bartosz-witkowski) [inline src/main/java/com/traveltime/sdk/dto/requests/TimeFilterProtoDistanceRequest.java:17]: I'm confused why we need a separate type for this

Comment (danielnaumau) [reply src/main/java/com/traveltime/sdk/dto/requests/TimeFilterProtoDistanceRequest.java:17]: Cause it has its own host, countries and transportation modes

File: src/main/java/com/traveltime/sdk/dto/requests/protodistance/Transportation.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum Transportation {
+    DRIVING_FERRY("driving+ferry", 3),
+    WALKING("walking", 4),
+    DRIVING("driving", 1);
+
```
Comment (Donatas-L) [inline src/main/java/com/traveltime/sdk/dto/requests/protodistance/Transportation.java:12]: Right now we only have 
```
driving+ferry
walking+ferry
```
You can't separate just driving or walking, right @bartosz-witkowski?

Comment (bartosz-witkowski) [reply src/main/java/com/traveltime/sdk/dto/requests/protodistance/Transportation.java:12]: We can and we even have maps for it - they're just not deployed.

Comment (danielnaumau) [reply src/main/java/com/traveltime/sdk/dto/requests/protodistance/Transportation.java:12]: should I leave them?

Comment (Donatas-L) [reply src/main/java/com/traveltime/sdk/dto/requests/protodistance/Transportation.java:12]: We don't have it for distance, please leave only `driving+ferry` and `walking+ferry`. If we add them in the future we will update SDK.

Comment (danielnaumau) [reply src/main/java/com/traveltime/sdk/dto/requests/protodistance/Transportation.java]: done

**Review by Donatas-L — APPROVED** [review]

**Pull request merged**